### PR TITLE
Book List & Auth header

### PR DESCRIPTION
**Pull request by miguelhenaoa**
# Summary
- List books from service
- Created component Book
- Created component BookDetails
- Created Token interceptor
- Add tests to Book and BookDetails components

# Trello Card
https://trello.com/c/EGz68ljL

# Screenshot
Captura de Pantalla 2021-06-08 a la(s) 3 55 26 p  m
Captura de Pantalla 2021-06-08 a la(s) 3 58 03 p  m
Captura de Pantalla 2021-06-08 a la(s) 3 58 34 p  m

### CODE REVIEW

File: src/app/components/book/book.component.html [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<div class="card">
+  <img [src]="book.image_url" alt="" class="book-cover">
```
Comment (ebedoyaalzate) [inline src/app/components/book/book.component.html]: add alt text

File: src/app/components/book/book.component.html [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+  <p class="title">{{book.title}}</p>
+  <p class="author">{{book.author}}</p>
+  <ul class="btn-add"> <span>+</span> </ul>
```
Comment (ebedoyaalzate) [inline src/app/components/book/book.component.html]: Is necessary the span tag?

File: src/app/components/book/book.component.scss [hunk 3]
```diff
@@ -0,0 +1,61 @@
+@import "src/styles/lib/colors";
+
+.card {
+  background-color: $white;
+  box-shadow: 4px 4px 10px 0 rgba(0, 0, 0, 0.5);
```
Comment (ebedoyaalzate) [inline src/app/components/book/book.component.scss]: use a color variable

File: src/app/components/book/book.component.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+@Component({
+  selector: 'app-book',
+  templateUrl: './book.component.html',
+  styleUrls: ['./book.component.scss']
+})
+export class BookComponent implements OnInit {
+  @Input() book!: Book;
+  constructor() { }
+
+  ngOnInit(): void {
```
Comment (ebedoyaalzate) [inline src/app/components/book/book.component.ts]: Delete methods if you don't use

File: src/app/helpers/guards/auth.guard.spec.ts [hunk 5]
```diff
@@ -6,7 +6,7 @@ import { UserService } from 'src/app/services/user.service';
 
 import { AuthGuard } from './auth.guard';
 
-describe('AuthGuard', () => {
+fdescribe('AuthGuard', () => {
```
Comment (ebedoyaalzate) [inline src/app/helpers/guards/auth.guard.spec.ts]: Delete this f in describe

File: src/app/screens/auth/book-details/book-details.component.html [hunk 6]
```diff
@@ -0,0 +1,19 @@
+<div class="content">
+  <a class="back" [routerLink]="'/auth/books'">
+    <img src="/assets/images/back-arrow.png" alt="">
```
Comment (ebedoyaalzate) [inline src/app/screens/auth/book-details/book-details.component.html]: add alt text

File: src/app/screens/auth/book-details/book-details.component.scss [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  height: 100vh;
+  width: 100vw;
+}
+
+.details-content {
+  background-color: $white;
+  box-shadow: 4px 4px 10px 0 rgba(0,0,0,0.5);
```
Comment (ebedoyaalzate) [inline src/app/screens/auth/book-details/book-details.component.scss]: Use a color variable

File: src/app/screens/auth/book-details/book-details.component.scss [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+        height: 4px;
+        left: 0;
+        position: absolute;
+        width: 100%;
+      }
+    }
+  }
+}
+
+p {
```
Comment (ebedoyaalzate) [inline src/app/screens/auth/book-details/book-details.component.scss]: Try to not use element selectors.

File: src/app/screens/auth/book-details/book-details.component.scss [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+  color: #828282;
+  font-size: 20px;
+  margin-top: 35px;
+
+  span {
```
Comment (ebedoyaalzate) [inline src/app/screens/auth/book-details/book-details.component.scss]: Same here

File: src/app/screens/auth/book-list/book-list.component.scss [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+}
+
+.search {
+  display: flex;
+  flex-flow: row nowrap;
+  justify-content: center;
+
+  .search-content {
+    align-items: center;
+    background-color: #fff;
```
Comment (ebedoyaalzate) [inline src/app/screens/auth/book-list/book-list.component.scss]: Use variables

File: src/app/services/books.service.ts [hunk 11]
```diff
@@ -0,0 +1,22 @@
...
+
+  private readonly apiBaseUrl = environment.gateway;
+
+  constructor(private http: HttpClient) { }
+
+  getBooks(): Observable<BooksResponse> {
+    return this.http.get<BooksResponse>(`${this.apiBaseUrl}/books`);
+  }
+
+  getBook(id: any): Observable<any> {
```
Comment (ebedoyaalzate) [inline src/app/services/books.service.ts]: Add interface

**Review by ebedoyaalzate — CHANGES_REQUESTED** [review]

File: src/app/components/book/book.component.spec.ts [hunk 12]
```diff
@@ -0,0 +1,27 @@
...
+    fixture = TestBed.createComponent(BookComponent);
+    component = fixture.componentInstance;
+    component.book = BookMock;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
+    expect(component).toBeTruthy();
+  });
+});
```
Comment (ebedoyaalzate) [inline src/app/components/book/book.component.spec.ts:36]: Add the respective tests

File: src/app/helpers/mocks/books.mock.ts [hunk 13]
```diff
@@ -0,0 +1,93 @@
...
+    }
+  ],
+  count: 1,
+  total_pages: 1,
+  total_count: 14,
+  current_page: 1,
+  next_page: 0
+};
+
+export const BooksFiltered = [
```
Comment (ebedoyaalzate) [inline src/app/helpers/mocks/books.mock.ts]: Add interface

File: src/app/helpers/mocks/books.mock.ts [hunk 14]
```diff
@@ -0,0 +1,93 @@
...
+    image_url: "https://covers.openlibrary.org/b/id/476938-L.jpg",
+    editor: "Pocket",
+    year: "8",
+    genre: "Sin registro",
+    created_at: new Date("2020-05-07T03:05:01.219Z"),
+    updated_at: new Date("2020-05-07T03:05:01.219Z")
+  }
+];
+
+export const BookMockResponse = {
```
Comment (ebedoyaalzate) [inline src/app/helpers/mocks/books.mock.ts]: Same here.

**Review by ebedoyaalzate — CHANGES_REQUESTED** [review]

File: src/app/components/book/book.component.spec.ts (hunk 12 above)
Comment (miguelhenaoa) [reply src/app/components/book/book.component.spec.ts:36]: This component not have methods

Comment (ebedoyaalzate) [reply src/app/components/book/book.component.spec.ts:36]: You could test the view in this component, the book that you add in the input shows well in the html

**Review by ebedoyaalzate — CHANGES_REQUESTED** [review]

File: src/app/components/book/book.component.html [hunk 15]
```diff
@@ -0,0 +1,6 @@
+<div class="card">
+  <img [src]="book.image_url" alt="Book cover" class="book-cover">
+  <p class="title">{{book.title}}</p>
+  <p class="author">{{book.author}}</p>
+  <button class="btn-add">+</button>
```
Comment (CrisVega08) [inline src/app/components/book/book.component.html:5]: And the on click event?

File: src/app/components/book/book.component.scss [hunk 16]
```diff
@@ -0,0 +1,56 @@
...
+  .btn-add {
+    align-items: center;
+    background-color: $atlantis;
+    border: hidden;
+    border-radius: 12px;
+    bottom: 5px;
+    color: $white;
+    display: flex;
+    font-size: 28px;
+    height: 24px;
```
Comment (CrisVega08) [inline src/app/components/book/book.component.scss]:
```suggestion
    height: 25px;
```

File: src/app/components/book/book.component.scss [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+    justify-content: center;
+    position: absolute;
+    right: 5px;
+    width: 24px;
```
Comment (CrisVega08) [inline src/app/components/book/book.component.scss]:
```suggestion
    width: 25px;
```

File: src/app/components/book/book.module.ts [hunk 18]
```diff
@@ -0,0 +1,16 @@
+import { NgModule } from '@angular/core';
+import { CommonModule } from '@angular/common';
+import { BookComponent } from './book.component';
+
+
```
Comment (CrisVega08) [inline src/app/components/book/book.module.ts]: remove

File: src/app/components/book/book.component.ts [hunk 19]
```diff
@@ -0,0 +1,12 @@
...
+import { Book } from 'src/app/screens/auth/interfaces/book.interface';
+
+@Component({
+  selector: 'app-book',
+  templateUrl: './book.component.html',
+  styleUrls: ['./book.component.scss']
+})
+export class BookComponent {
+  @Input() book!: Book;
+
```
Comment (CrisVega08) [inline src/app/components/book/book.component.ts]: remove

File: src/app/screens/auth/book-details/book-details.component.html [hunk 20]
```diff
@@ -0,0 +1,19 @@
+<div class="content">
+  <button class="back" [routerLink]="'/auth/books'">
```
Comment (CrisVega08) [inline src/app/screens/auth/book-details/book-details.component.html]: this is not a button use <a>

**Review by CrisVega08 — CHANGES_REQUESTED** [review]

File: src/app/components/book/book.component.html (hunk 15 above)
Comment (miguelhenaoa) [reply src/app/components/book/book.component.html:5]: the on click event is added to this button in the following branch

**Review by ebedoyaalzate — APPROVED** [review]

**Review by CrisVega08 — APPROVED** [review]

**Pull request merged**